microservices/front-end/src/components/Register.tsx

```diff
@@ -58,19 +61,25 @@ const Register: React.FC = () => {
       });
 
       if (response.status == 200) {
-        navigate("/login");
+        setSuccess("Account successfully created!");
+
+        setTimeout(() => {
+          navigate("/login");
+        }, 500); // 500ms delay before navigating, let user see success message
```

Comment:
I think we can have a snack bar component + "snack bar context" that shows even between page loads so that we don't have to resort to using `setTimeout` (especially for success messages)

(see [https://mui.com/material-ui/react-snackbar/#consecutive-snackbars](https://mui.com/material-ui/react-snackbar/#consecutive-snackbars))

Note: imo 500ms is a little short for the user to read the message fully

Alternatively, when the success message shows, the text fields and buttons should be disabled to prevent the user from clicking and submitting forms twice within that 500ms window

Reply:
Hi I have added the status check to disable the "save" and "sign up" button when the submission process is ongoing, to prevent user from clicking it multiple times. This is the suboptimal choice, and I will look into the first choice more when I finished my midterms.